src/core/src/UrlAssetAccessor.cpp

```diff
@@ -0,0 +1,268 @@
+// Copyright(c) 2023 Timothy Moore
+// Copyright 2024 CesiumGS, Inc. and Contributors
+
+#include "cesium/omniverse/UrlAssetAccessor.h"
+
+#include <CesiumAsync/IAssetResponse.h>
+#include <CesiumUtility/Tracing.h>
+#include <omni/kit/IApp.h>
+
+#include <algorithm>
+#include <cstring>
+
+namespace cesium::omniverse {
+const auto CURL_BUFFERSIZE = 3145728L; // 3 MiB
+
+class UrlAssetResponse : public CesiumAsync::IAssetResponse {
+  public:
+    uint16_t statusCode() const override {
+        return _statusCode;
+    }
+
+    std::string contentType() const override {
+        return _contentType;
+    }
+
+    const CesiumAsync::HttpHeaders& headers() const override {
+        return _headers;
+    }
+
+    gsl::span<const std::byte> data() const override {
+        return {const_cast<const std::byte*>(_result.data()), _result.size()};
+    }
+
+    static size_t headerCallback(char* buffer, size_t size, size_t nitems, void* userData);
+    static size_t dataCallback(char* buffer, size_t size, size_t nitems, void* userData);
+    void setCallbacks(CURL* curl);
+    uint16_t _statusCode = 0;
+    std::string _contentType;
+    CesiumAsync::HttpHeaders _headers;
+    std::vector<std::byte> _result;
+};
+
+class UrlAssetRequest : public CesiumAsync::IAssetRequest {
+  public:
+    UrlAssetRequest(std::string method, std::string url, CesiumAsync::HttpHeaders headers)
+        : _method(std::move(method))
+        , _url(std::move(url))
+        , _headers(std::move(headers)) {}
+
+    UrlAssetRequest(
+        std::string method,
+        std::string url,
+        const std::vector<CesiumAsync::IAssetAccessor::THeader>& headers)
+        : _method(std::move(method))
+        , _url(std::move(url)) {
+        _headers.insert(headers.begin(), headers.end());
+
+        const auto app = carb::getCachedInterface<omni::kit::IApp>();
+        const auto& buildInfo = app->getBuildInfo();
+        const auto platformInfo = app->getPlatformInfo();
+
+        _headers.insert({"X-Cesium-Client", "Cesium for Omniverse"});
+        _headers.insert(
+            {"X-Cesium-Client-Version", fmt::format("v{} {}", CESIUM_OMNI_VERSION, CESIUM_OMNI_GIT_HASH_ABBREVIATED)});
+        _headers.insert({"X-Cesium-Client-Engine", fmt::format("Kit SDK {}", buildInfo.kitVersion)});
+        _headers.insert({"X-Cesium-Client-OS", platformInfo.platform});
+    }
+
+    const std::string& method() const override {
+        return this->_method;
```

Comment:
Style nitpick: convention in this repo is to omit `this->`. Though looks like native is different.

Reply:
I prefer to omit `this->` too. I probably copied the code directly from cesium-unreal.